### ISSUE TITLE
What if I only want the part below the nose for the mouth drive，like  this？

### ISSUE THREAD

**Issue by zhanghongyong123456** [willyfh/farl-face-segmentation#1]
![image](https://github.com/willyfh/farl-face-segmentation/assets/48466610/24c2764d-31bf-4708-a151-53e0ccaa6d85)

**Comment by willyfh** [comment]
Hello, thanks for asking! If you're looking to grab a specific part from an image (as illustrated) without diving into model training or fine-tuning, you may want to consider using face landmark detection with MediaPipe (you can find the details [here](https://developers.google.com/mediapipe/solutions/vision/face_landmarker)). Once you've spotted the landmarks, you can easily snatch the area you're after within the desired points.

Example of facial landmarks detected by MediaPipe:

![image](https://github.com/willyfh/farl-face-segmentation/assets/5786636/a33289da-866c-4319-a866-fb594a419fa7)

**Comment by zhanghongyong123456** [comment]
> Hello, thanks for asking! If you're looking to grab a specific part from an image (as illustrated) without diving into model training or fine-tuning, you may want to consider using face landmark detection with MediaPipe (you can find the details [here](https://developers.google.com/mediapipe/solutions/vision/face_landmarker)). Once you've spotted the landmarks, you can easily snatch the area you're after within the desired points.
> 
> Example of facial landmarks detected by MediaPipe:
> 
> ![image](https://private-user-images.githubusercontent.com/5786636/289402251-a33289da-866c-4319-a866-fb594a419fa7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDIzNDYzMTEsIm5iZiI6MTcwMjM0NjAxMSwicGF0aCI6Ii81Nzg2NjM2LzI4OTQwMjI1MS1hMzMyODlkYS04NjZjLTQzMTktYTg2Ni1mYjU5NGE0MTlmYTcucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQUlXTkpZQVg0Q1NWRUg1M0ElMkYyMDIzMTIxMiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyMzEyMTJUMDE1MzMxWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZWI2YWFlMWE5OTZlZjIyYTkxODAzNTgyZWZmZDk1MGRlMWM4OTgwN2ZjY2I2YTk3NDM0N2ZhMWVjNDU2NTQ1OCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.jp5d9JuYipAd6TUVtt4D8c-WplGvQk5e_uV937CMhyA)

Thank you very much for your answer. I want to get the mask of this, which seems not to be a simple face mask,Do you have a good way to get it

https://stylelipsync.github.io/

![image](https://github.com/willyfh/farl-face-segmentation/assets/48466610/d5c569c1-5464-4a5e-b3e7-04aba862fc84)

**Comment by willyfh** [comment]
> > Hello, thanks for asking! If you're looking to grab a specific part from an image (as illustrated) without diving into model training or fine-tuning, you may want to consider using face landmark detection with MediaPipe (you can find the details [here](https://developers.google.com/mediapipe/solutions/vision/face_landmarker)). Once you've spotted the landmarks, you can easily snatch the area you're after within the desired points.
> > Example of facial landmarks detected by MediaPipe:
> > ![image](https://private-user-images.githubusercontent.com/5786636/289402251-a33289da-866c-4319-a866-fb594a419fa7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDIzNDYzMTEsIm5iZiI6MTcwMjM0NjAxMSwicGF0aCI6Ii81Nzg2NjM2LzI4OTQwMjI1MS1hMzMyODlkYS04NjZjLTQzMTktYTg2Ni1mYjU5NGE0MTlmYTcucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQUlXTkpZQVg0Q1NWRUg1M0ElMkYyMDIzMTIxMiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyMzEyMTJUMDE1MzMxWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZWI2YWFlMWE5OTZlZjIyYTkxODAzNTgyZWZmZDk1MGRlMWM4OTgwN2ZjY2I2YTk3NDM0N2ZhMWVjNDU2NTQ1OCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.jp5d9JuYipAd6TUVtt4D8c-WplGvQk5e_uV937CMhyA)
> 
> Thank you very much for your answer. I want to get the mask of this, which seems not to be a simple face mask,Do you have a good way to get it
> 
> https://stylelipsync.github.io/
> 
> ![image](https://private-user-images.githubusercontent.com/48466610/289721102-d5c569c1-5464-4a5e-b3e7-04aba862fc84.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDMzMzQ5NTEsIm5iZiI6MTcwMzMzNDY1MSwicGF0aCI6Ii80ODQ2NjYxMC8yODk3MjExMDItZDVjNTY5YzEtNTQ2NC00YTVlLWIzZTctMDRhYmE4NjJmYzg0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjIzVDEyMzA1MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTg2NDBiMmM0ODE0M2FhNGNlMmM4OWY2MGViMjU1Y2Y3M2U3MzQwNTVkY2Q3OTk4YTA0MTFiZGY0YTMzZjNiM2MmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.Kevhko6HUDpw9y8-_7MetjG71So6RpoEgfaGm63mRog)

Hi, if you specifically referring to the 3D Parametric Mesh Predictor, I am not quite sure about it. I am actually not an expert in the field of face analysis. I tried to quickly read the given reference. It seems that the paper specifically mentioned this project (https://github.com/microsoft/Deep3DFaceReconstruction) as their 3D face mesh predictor. As you may already know, maybe you can try to refer to Subsection 3.1 in the paper ([here](https://openaccess.thecvf.com/content/ICCV2023/papers/Ki_StyleLipSync_Style-based_Personalized_Lip-sync_Video_Generation_ICCV_2023_paper.pdf)) to construct the masking.
